packages/proxy-agent/src/index.ts

```diff
@@ -41,7 +39,7 @@ export const proxies: {
 	'pac-ftp': [PacProxyAgent, PacProxyAgent],
 	'pac-http': [PacProxyAgent, PacProxyAgent],
 	'pac-https': [PacProxyAgent, PacProxyAgent],
-};
+} satisfies { [P in ValidProtocol]: [AgentConstructor, AgentConstructor] };
```

Comment:
Using the [`satisfies` operator introduced in TS 4.9](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-4-9.html#the-satisfies-operator) allows us to preserve more specific type information on the `proxies` object. That means that `// @ts-expect-error meh…` below is no longer necessary, I was able to remove it 😄

Reply:
Nice! This _may_ constitute a breaking change though. If we're relying on any TS 4.9+ features already then that wouldn't be the case.

Reply:
Oh interesting, I hadn't thought of that, just assumed it would be OK because this repo uses TS 5.0.

I just checked the contents of `packages/proxy-agent/dist/index.d.ts`, and it looks like `satisfies` operator has been stripped out of the compiled type definition:

```ts
/**
 * Supported proxy types.
 */
export declare const proxies: {
    http: [typeof HttpProxyAgent, typeof HttpsProxyAgent];
    https: [typeof HttpProxyAgent, typeof HttpsProxyAgent];
    socks: [typeof SocksProxyAgent, typeof SocksProxyAgent];
    socks4: [typeof SocksProxyAgent, typeof SocksProxyAgent];
    socks4a: [typeof SocksProxyAgent, typeof SocksProxyAgent];
    socks5: [typeof SocksProxyAgent, typeof SocksProxyAgent];
    socks5h: [typeof SocksProxyAgent, typeof SocksProxyAgent];
    'pac-data': [typeof PacProxyAgent, typeof PacProxyAgent];
    'pac-file': [typeof PacProxyAgent, typeof PacProxyAgent];
    'pac-ftp': [typeof PacProxyAgent, typeof PacProxyAgent];
    'pac-http': [typeof PacProxyAgent, typeof PacProxyAgent];
    'pac-https': [typeof PacProxyAgent, typeof PacProxyAgent];
};
```

I guess operators like this are just used at compile time. TIL!

Reply:
<picture data-single-emoji=":til:" title=":til:"><img class="emoji" src="https://emoji.slack-edge.com/T0CAQ00TU/til/d5952483dda1dda3.png" alt=":til:" width="20" height="auto" align="absmiddle"></picture> as well!